core/src/main/scala/org/apache/spark/deploy/rest/StandaloneRestClient.scala

```diff
@@ -293,6 +355,17 @@ private[deploy] class StandaloneRestClient extends Logging {
   private def handleUnexpectedRestResponse(unexpected: SubmitRestProtocolResponse): Unit = {
     logError(s"Error: Server responded with message of unexpected type ${unexpected.messageType}.")
   }
+
+  /**
+   * When a connection exception was caught, we see whether all masters are lost.
```

Comment:
```
/**
 * When a connection exception is caught, return true if all masters are lost.
 * Note that the heuristic used here does not take into account that masters
 * can recover during the lifetime of this client. This assumption should be
 * harmless because this client currently does not support retrying submission
 * on failure yet (SPARK-6443).
 */
```